PR: https://github.com/odoo/odoo/pull/

From: 8553c3f6483e4b89c3a0b707addbe48fec0ec12f
From: Srushti Patel
Date: 2016-11-10 12:22:36

Structural Changes: 8
Total Changes: 8

[IMP] stock: add tooltips on push and procurement rules

Push and procurement rules are some difficult concepts to understand.

Add some tooltips :

Push Rules tooltip :
When a move is foreseen to a location, the push rule will automatically create a move to a next location after. This is mainly only needed when creating manual operations e.g. 2/3 step manual purchase order or 2/3 step finished product manual manufacturing order. In other cases, it is important to use pull rules where you know where you are going based on a demand.

Procurement Rules :
The demand represented by a procurement from e.g. a sale order, a reordering rule, another move, ... needs to be solved by applying a procurement rule. Depending on the action on the procurement rule, this triggers a purchase order, manufacturing order or another move. This way we create chains in the reverse order from the endpoint with the original demand to the starting point. That way, it is always known where we need to go and that is why they are preferred over push rules.

================================= pseudo patch: =================================

--- a/addons/stock/models/stock_location.py
+++ b/addons/stock/models/stock_location.py
@@ -111,8 +111,12 @@ class Route(models.Model):
     name = fields.Char('Route Name', required=True, translate=True)
     active = fields.Boolean('Active', default=True, help="If the active field is set to False, it will allow you to hide the route without removing it.")
     sequence = fields.Integer('Sequence', default=0)
-    pull_ids = fields.One2many('procurement.rule', 'route_id', 'Procurement Rules', copy=True)
-    push_ids = fields.One2many('stock.location.path', 'route_id', 'Push Rules', copy=True)
+    pull_ids = fields.One2many('procurement.rule', 'route_id', 'Procurement Rules', copy=True, 
+        help="The demand represented by a procurement from e.g. a sale order, a reordering rule, another move, needs to be solved by applying a procurement rule. Depending on the action on the procurement rule,"\
+        "this triggers a purchase order, manufacturing order or another move. This way we create chains in the reverse order from the endpoint with the original demand to the starting point. "\
+        "That way, it is always known where we need to go and that is why they are preferred over push rules.")
+    push_ids = fields.One2many('stock.location.path', 'route_id', 'Push Rules', copy=True, 
+        help="When a move is foreseen to a location, the push rule will automatically create a move to a next location after. This is mainly only needed when creating manual operations e.g. 2/3 step manual purchase order or 2/3 step finished product manual manufacturing order. In other cases, it is important to use pull rules where you know where you are going based on a demand.")
     product_selectable = fields.Boolean('Applicable on Product', default=True, help="When checked, the route will be selectable in the Inventory tab of the Product form.  It will take priority over the Warehouse route. ")
     product_categ_selectable = fields.Boolean('Applicable on Product Category', help="When checked, the route will be selectable on the Product Category.  It will take priority over the Warehouse route. ")
     warehouse_selectable = fields.Boolean('Applicable on Warehouse', help="When a warehouse is selected for this route, this route should be seen as the default route when products pass through this warehouse.  This behaviour can be overridden by the routes on the Product/Product Categories or by the Preferred Routes on the Procurement")
